internal/resources/selfmonitor/resources.go

```diff
@@ -295,15 +295,23 @@ func makePodSpec(baseName, image string, opts ...podSpecOption) corev1.PodSpec {
 					},
 				},
 				VolumeMounts: []corev1.VolumeMount{{Name: "prometheus-config-volume", MountPath: "/etc/prometheus/"}, {Name: "prometheus-storage-volume", MountPath: "/prometheus/"}},
+				Ports: []corev1.ContainerPort{
+					{
+						Name:          "http-web",
+						ContainerPort: ports.PrometheusPort,
+					},
+				},
 				LivenessProbe: &corev1.Probe{
 					ProbeHandler: corev1.ProbeHandler{
-						HTTPGet: &corev1.HTTPGetAction{Path: "/-/ready", Port: intstr.IntOrString{IntVal: ports.PrometheusPort}},
+						HTTPGet: &corev1.HTTPGetAction{Path: "/-/healthy", Port: intstr.IntOrString{IntVal: ports.PrometheusPort}},
 					},
+					FailureThreshold: 5, PeriodSeconds: 5, TimeoutSeconds: 3, SuccessThreshold: 1,
 				},
 				ReadinessProbe: &corev1.Probe{
 					ProbeHandler: corev1.ProbeHandler{
 						HTTPGet: &corev1.HTTPGetAction{Path: "/-/ready", Port: intstr.IntOrString{IntVal: ports.PrometheusPort}},
 					},
+					FailureThreshold: 3, PeriodSeconds: 5, TimeoutSeconds: 3, SuccessThreshold: 1,
```

Comment:
The readiness/liveness probes deviate from the ones in the community chart: https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/values.yaml#L595. Maybe also add a startup probe?

Reply:
In the community chart, startup probe is disabled by default, I think not required in our case because we have a light-weight version of prometheus and we don't have slow startup in our deployment